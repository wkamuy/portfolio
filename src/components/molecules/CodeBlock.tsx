// ~/components/CodeBlock.tsx
import { FC } from 'react';
import { Prism as SyntaxHighlighter } from 'react-syntax-highlighter';
import materialDark from 'react-syntax-highlighter/dist/cjs/styles/prism/material-dark';

type P = {
  value: string;
  language?: string;
};

const CodeBlock: FC<P> = ({ value, language }) => (
  // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
  <SyntaxHighlighter language={language} style={materialDark}>
    {value}
  </SyntaxHighlighter>
);

export default CodeBlock;
