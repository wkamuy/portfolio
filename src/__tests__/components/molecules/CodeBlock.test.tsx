import * as React from 'react';
import * as renderer from 'react-test-renderer';
import CodeBlock from '../../../components/molecules/CodeBlock';

describe('<CodeBlock />', () => {
  it('sould render correctly', () => {
    const value = 'test';
    const tree = renderer.create(<CodeBlock value={value} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
