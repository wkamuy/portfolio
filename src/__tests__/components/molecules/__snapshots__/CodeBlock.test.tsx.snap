// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CodeBlock /> sould render correctly 1`] = `
<pre
  style={
    Object {
      "MozHyphens": "none",
      "MozTabSize": "4",
      "OTabSize": "4",
      "WebkitHyphens": "none",
      "background": "#2f2f2f",
      "color": "#eee",
      "fontFamily": "Roboto Mono, monospace",
      "fontSize": "1em",
      "hyphens": "none",
      "lineHeight": "1.5em",
      "margin": "0.5em 0",
      "msHyphens": "none",
      "overflow": "auto",
      "padding": "1.25em 1em",
      "position": "relative",
      "tabSize": "4",
      "textAlign": "left",
      "whiteSpace": "pre",
      "wordBreak": "normal",
      "wordSpacing": "normal",
      "wordWrap": "normal",
    }
  }
>
  <code
    style={
      Object {
        "MozHyphens": "none",
        "MozTabSize": "4",
        "OTabSize": "4",
        "WebkitHyphens": "none",
        "background": "#2f2f2f",
        "color": "#eee",
        "fontFamily": "Roboto Mono, monospace",
        "fontSize": "1em",
        "hyphens": "none",
        "lineHeight": "1.5em",
        "msHyphens": "none",
        "tabSize": "4",
        "textAlign": "left",
        "whiteSpace": "pre",
        "wordBreak": "normal",
        "wordSpacing": "normal",
        "wordWrap": "normal",
      }
    }
  >
    <span
      style={Object {}}
    >
      test
    </span>
  </code>
</pre>
`;
